### DISCUSSION TITLE
Conditionally rendering cell data based on accessor value

### DISCUSSION THREAD

**Discussion by tkrebs2** [TanStack/table#2093] [General] [answered]
Just started using react-table, I have a lot of cases where I need to render certain things in table cells based on their values.

So for example, 
`{
     Header: "Action",
     id: "startTime", 
     accessor: "attributes.startTime"
   }`

If `attributes.startTime === null` I want to render a button `<button .. > some action </button`.
I have a [codesandbox here](https://codesandbox.io/s/react-table-executions-6gjy1) to make my use case more clear.

**Comment by tannerlinsley** [comment, marked as answer]
Your sandbox looks great to me. Using a ternary in the Cell renderer to check the value and render accordingly is the best way to achieve this! :)

**Reply by tkrebs2** [reply]
What exactly is `props.value`? It seems like the ternary is always evaluating to true or false.

**Reply by tkrebs2** [reply]
```          {
            Header: "Action",
            id: "startTime",
            accessor: "attributes.startTime",
            Cell: props => {
              return props.value == null ? (
                <button>Click Me </button>
              ) : (
                <button disabled> No action </button>
              );
            }
          }```

**Reply by tannerlinsley** [reply]
`props.value` is the cell's value that was accessed from `row => row.attributes.startTime`

**Reply by tkrebs2** [reply]
hmm something is off with the comparison. Half the `attributes.startTime` values are null while the other half contains date strings, but it's still rendering as if they are all `null`

**Reply by tannerlinsley** [reply]
Dates are interesting. Aside from checking if it's null or not, I would suggest something like this:

```js
Header: "Action",
            id: "startTime",
            accessor: row => row.attributes.startTime ? new Date(row.attributes.startTime) : null,
            Cell: props => {
// You'll want to format your date here too, something like props.value.toLocaleDateString()
              return props.value == null ? (
                <button>Click Me </button>
              ) : (
                <button disabled> No action </button>
              );
            }
```

**Reply by tkrebs2** [reply]
Thanks for the feedback. The weird thing is I don't actually want to manipulate the dates in this cell, I just want to simply check if something other than an empty string or null value exists in `attributes.startTime`.  For some perspective, I want to render a cancel button for the objects (schedules) that do have a date string inside `attributes.startTime`, no actual formatting is needed, that is handled in other cells.

**Reply by tannerlinsley** [reply]
Try this: https://codesandbox.io/s/react-table-executions-h376q

**Reply by tkrebs2** [reply]
Yup exactly what I was looking for. Thank you very much.
